packages/aws-cdk-lib/aws-lambda/lib/runtime.ts

```diff
@@ -56,43 +62,64 @@ export class Runtime {
    * The NodeJS runtime (nodejs)
    * @deprecated Legacy runtime no longer supported by AWS Lambda. Migrate to the latest NodeJS runtime.
    */
-  public static readonly NODEJS = new Runtime('nodejs', RuntimeFamily.NODEJS, { supportsInlineCode: true });
+  public static readonly NODEJS = new Runtime('nodejs', RuntimeFamily.NODEJS, {
+    supportsInlineCode: true,
+    isDeprecated: true,
+  });
 
   /**
    * The NodeJS 4.3 runtime (nodejs4.3)
    * @deprecated Legacy runtime no longer supported by AWS Lambda. Migrate to the latest NodeJS runtime.
    */
-  public static readonly NODEJS_4_3 = new Runtime('nodejs4.3', RuntimeFamily.NODEJS, { supportsInlineCode: true });
+  public static readonly NODEJS_4_3 = new Runtime('nodejs4.3', RuntimeFamily.NODEJS, {
+    supportsInlineCode: true,
+    isDeprecated: true,
+  });
 
   /**
    * The NodeJS 6.10 runtime (nodejs6.10)
    * @deprecated Legacy runtime no longer supported by AWS Lambda. Migrate to the latest NodeJS runtime.
    */
-  public static readonly NODEJS_6_10 = new Runtime('nodejs6.10', RuntimeFamily.NODEJS, { supportsInlineCode: true });
+  public static readonly NODEJS_6_10 = new Runtime('nodejs6.10', RuntimeFamily.NODEJS, {
+    supportsInlineCode: true,
+    isDeprecated: true,
+  });
 
   /**
    * The NodeJS 8.10 runtime (nodejs8.10)
    * @deprecated Legacy runtime no longer supported by AWS Lambda. Migrate to the latest NodeJS runtime.
    */
-  public static readonly NODEJS_8_10 = new Runtime('nodejs8.10', RuntimeFamily.NODEJS, { supportsInlineCode: true });
+  public static readonly NODEJS_8_10 = new Runtime('nodejs8.10', RuntimeFamily.NODEJS, {
+    supportsInlineCode: true,
+    isDeprecated: true,
```

Comment:
If you're going to add this, also add a check to `Function` to add a `Annotations.of(this).addWarningV2(...)` that tells people the runtime is deprecated.